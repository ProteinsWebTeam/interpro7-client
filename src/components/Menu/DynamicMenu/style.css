.dynamic-menu i {
  display: none;
}
