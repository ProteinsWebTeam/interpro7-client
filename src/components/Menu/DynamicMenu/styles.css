@import '../../../styles/timing.css';
@import '../../../styles/colors.css';
@import '../../../styles/spacing.css';

.menu {
  display: flex;
}

.menu-item {
  pointer-events: none;
  opacity: 0;
  transition: opacity var(--timing-fast) ease-in-out;
  border-left: 1px solid #fff3;

  &:first-child {
    border-left: none;
  }
}

.more {
  position: absolute;
  right: 0;
  height: 2rem;
  margin: 0 1ch;
  cursor: pointer;
  text-align: end;
  border: none;
  display: flex;
  flex-direction: column;

  & .menu-item {
    background: var(--colors-main);
    list-style: none;
    pointer-events: none;
    opacity: 0;
    transition: opacity var(--timing-fast) ease-in-out;
  }

  & a {
    display: block;
    padding: var(--spacing-small) var(--spacing-medium);
    text-align: center;
  }

  &:active,
  &:focus,
  &:hover {
    & .menu-item {
      pointer-events: initial;
      opacity: 1;
    }
  }
}

.visible {
  pointer-events: initial;
  opacity: 1;
}
