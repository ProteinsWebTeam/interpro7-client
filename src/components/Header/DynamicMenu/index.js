// @flow
import React, {PropTypes as T} from 'react';
import {withRouter} from 'react-router/es';
import {connect} from 'react-redux';

import MenuItem from 'components/Menu/MenuItem';

import {foundationPartial} from 'styles/foundation';
import ebiStyles from 'styles/ebi-global.css';
import interproStyles from 'styles/interpro-new.css';

import {InterPro, entities, singleEntity} from 'menuConfig';

const styles = foundationPartial(ebiStyles, interproStyles);

const InterProMenu = ({pathname}) => (
  <ul className={styles('menu')}>
    {InterPro.map(({to, name}) => (
      <li key={to}>
        <MenuItem to={to} className={styles({active: pathname === to})}>
          {name}
        </MenuItem>
      </li>
    ))}
  </ul>
);
InterProMenu.propTypes = {
  pathname: T.string.isRequired,
};

const SingleEntityMenu = ({data, type, pathname}) => {
  const [baseURL] = pathname.match(
    new RegExp(`^.*${data.metadata.accession}`, 'i')
  );
  return (
    <ul className={styles('menu')}>
      {singleEntity
        .filter(({to}) => !to.includes(type))
        .map(({to, name, counter}) => (
          <li key={to}>
            <MenuItem
              to={baseURL + to}
              activeClassName={styles('active')}
              disabled={counter && !data.metadata.counters[counter]}
            >
              {name}&nbsp;
              {counter &&
                <span className={styles('badge')}>
                  {data.metadata.counters[counter] || 0}
                </span>
              }
            </MenuItem>
          </li>
        ))
      }
    </ul>
  );
};
SingleEntityMenu.propTypes = {
  data: T.object.isRequired,
  type: T.string.isRequired,
  pathname: T.string.isRequired,
};

const EntityMenu = () => (
  <ul className={styles('menu')}>
    {entities.map(({to, name}) => (
      <li key={to}>
        <MenuItem to={to} activeClassName={styles('active')}>{name}</MenuItem>
      </li>
    ))}
  </ul>
);

const DynamicMenu = ({data, location: {pathname}}) => {
  if (!data || pathname === '/') return <InterProMenu pathname={pathname} />;
  const type = pathname.match(/^\/?([^/]*)/)[1].toLowerCase();
  if (data.metadata) {
    return (
      <SingleEntityMenu data={data} type={type} pathname={pathname} />
    );
  }
  return <EntityMenu data={data} type={type} />;
};
DynamicMenu.propTypes = {
  data: T.object,
  location: T.shape({
    pathname: T.string.isRequired,
  }).isRequired,
};

export default withRouter(connect(({data: {data}}) => ({data}))(DynamicMenu));

