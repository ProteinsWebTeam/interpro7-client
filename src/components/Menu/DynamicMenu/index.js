// @flow
import React, { PureComponent } from 'react';
import T from 'prop-types';

import MenuItem from 'components/Menu/MenuItem';

import { schedule, sleep } from 'timing-functions';

import { InterPro } from 'menuConfig';

import { foundationPartial } from 'styles/foundation';

import fonts from 'EBI-Icon-fonts/fonts.css';
import interproStyles from 'styles/interpro-new.css';
import styles from './styles.css';

import dotsvg from 'images/icons/ico-more.svg';

const f = foundationPartial(fonts, interproStyles, styles);

const RECHECK_AFTER_MOUNT = 500; // ms
const MAX_DELAY_BEFORE_CHECKING_FIT = 200; // ms

const InterProMin = InterPro.filter((item) => item.name !== 'Settings');

const MenuItemWithEntities = (
  { to, activeClass, exact, name, entities, nested } /*: { 
      to: Object| function,
      name: string,   
      activeClass?: string | function,
      exact:boolean, 
      nested?:boolean, 
      entities?: Array<{
        to:{}, 
        name: string, 
        exact:boolean, 
      }>  
    } */,
) =>
  entities ? (
    <div className={f('dropdown')}>
      <MenuItem to={to} activeClass={activeClass} exact={exact}>
        <span className={f('arrow')}>â–¾</span> {name}
      </MenuItem>
      <ul role="tree" tabIndex="0" className={f({ nested })}>
        {entities.map(({ to, name, exact }) => (
          <li key={name} className={f('menu-item')} role="treeitem">
            <MenuItem to={to} activeClass={f('reactive')} exact={exact}>
              {name}
            </MenuItem>
          </li>
        ))}
      </ul>
    </div>
  ) : (
    <MenuItem to={to} activeClass={activeClass} exact={exact}>
      {name}
    </MenuItem>
  );

MenuItemWithEntities.propTypes = {
  exact: T.bool,
  name: T.string,
  entities: T.arrayOf(T.object),
  activeClass: T.oneOfType([T.string, T.func]),
  to: T.oneOfType([T.object, T.func]),
  nested: T.bool,
};
/*:: type Props = {
  width: number,
}; */
/*:: type State = {
  [key: string]: boolean,
}; */

class DynamicMenu extends PureComponent /*:: <Props, State> */ {
  static propTypes = {
    width: T.number.isRequired,
  };
  /*:: _menuItems: Set<HTMLElement>; */
  /*:: _dotdotdotRef: { current: null | React$ElementRef<'span'> }; */

  static defaultProps = {
    width: +Infinity,
  };

  constructor(props /*: Props */) {
    super(props);

    this.state = {};
    for (const { name } of InterProMin) {
      this.state[name] = true;
    }

    this._menuItems = new Set();
    this._dotdotdotRef = React.createRef();
  }

  async componentDidMount() {
    await sleep(RECHECK_AFTER_MOUNT);
    this._checkSizes(this.props.width);
  }

  componentDidUpdate() {
    this._checkSizes(this.props.width);
  }

  _checkSizes = async (width /*: number */) => {
    await schedule(MAX_DELAY_BEFORE_CHECKING_FIT);
    let remainingWidth =
      width - (this._dotdotdotRef.current?.getBoundingClientRect().width || 0);
    for (const menuItem of this._menuItems) {
      if (remainingWidth > 0) {
        const { width } = menuItem.getBoundingClientRect();
        remainingWidth -= width;
        // eslint-disable-next-line react/no-did-update-set-state
        this.setState({ [menuItem.dataset.name]: remainingWidth >= 0 });
      } else {
        // eslint-disable-next-line react/no-did-update-set-state
        this.setState({ [menuItem.dataset.name]: false });
      }
    }
  };

  _setMenuItemInMap = (node) => {
    if (node instanceof HTMLElement) this._menuItems.add(node);
  };

  render() {
    const hiddenItems = InterProMin.filter(({ name }) => !this.state[name]);
    return (
      <ul className={f('menu')}>
        {InterProMin.map(({ to, name, activeClass, exact, entities }) => (
          <li
            key={name}
            ref={this._setMenuItemInMap}
            data-name={name}
            className={f('menu-item', { visible: this.state[name] || false })}
            data-testid={`menu-tab-${name.toLowerCase().replace(/\s+/g, '_')}`}
          >
            <MenuItemWithEntities
              to={to}
              activeClass={activeClass}
              exact={exact}
              name={name}
              entities={entities}
            />
          </li>
        ))}
        <ul
          className={f('menu-item', 'view-more', {
            visible: hiddenItems.length,
          })}
          role="tree"
          tabIndex="0"
        >
          <span className={f('more-icon-container')} ref={this._dotdotdotRef}>
            <img src={dotsvg} width="30px" alt="view all menu items" />
          </span>

          {hiddenItems.map(({ to, name, activeClass, exact, entities }) => (
            <li key={name} className={f('menu-item')} role="treeitem">
              <MenuItemWithEntities
                to={to}
                activeClass={activeClass}
                exact={exact}
                name={name}
                entities={entities}
                nested={true}
              />
            </li>
          ))}
        </ul>
      </ul>
    );
  }
}

export default DynamicMenu;
