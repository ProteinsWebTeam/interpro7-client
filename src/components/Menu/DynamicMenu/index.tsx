import React, { PureComponent } from 'react';

import MenuItem from 'components/Menu/MenuItem';

import { schedule, sleep } from 'timing-functions';

import { InterPro, MenuItemProps } from 'menuConfig';

import cssBinder from 'styles/cssBinder';

import fonts from 'EBI-Icon-fonts/fonts.css';
import styles from './styles.css';

import dotsvg from 'images/icons/ico-more.svg';

const css = cssBinder(fonts, styles);

const RECHECK_AFTER_MOUNT = 500; // ms
const MAX_DELAY_BEFORE_CHECKING_FIT = 200; // ms

const InterProMin = InterPro.filter((item) => item.name !== 'Settings');

const MenuItemWithEntities = ({
  to,
  activeClass,
  exact,
  name,
  entities,
  nested,
}: MenuItemProps & { nested?: boolean }) =>
  entities ? (
    <div className={css('dropdown')}>
      <MenuItem to={to} activeClass={activeClass} exact={exact}>
        <span className={css('arrow')}>â–¾</span> {name}
      </MenuItem>
      <ul role="tree" tabIndex={0} className={css({ nested })}>
        {entities.map(({ to, name, exact }) => (
          <li
            key={name}
            className={css('menu-item')}
            role="treeitem"
            aria-selected="false"
          >
            <MenuItem to={to} activeClass={css('reactive')} exact={exact}>
              {name}
            </MenuItem>
          </li>
        ))}
      </ul>
    </div>
  ) : (
    <MenuItem to={to} activeClass={activeClass} exact={exact}>
      {name}
    </MenuItem>
  );

type Props = {
  width: number;
};
type State = {
  [key: string]: boolean;
};

class DynamicMenu extends PureComponent<Props, State> {
  private _menuItems: Set<HTMLElement>;
  private _dotdotdotRef: React.RefObject<HTMLSpanElement>;

  static defaultProps = {
    width: +Infinity,
  };

  constructor(props: Props) {
    super(props);

    const state: State = {};
    for (const { name } of InterProMin) {
      state[name] = true;
    }
    this.state = state;

    this._menuItems = new Set();
    this._dotdotdotRef = React.createRef();
  }

  async componentDidMount() {
    await sleep(RECHECK_AFTER_MOUNT);
    this._checkSizes(this.props.width);
  }

  componentDidUpdate() {
    this._checkSizes(this.props.width);
  }

  _checkSizes = async (width: number) => {
    await schedule(MAX_DELAY_BEFORE_CHECKING_FIT);
    let remainingWidth =
      width - (this._dotdotdotRef.current?.getBoundingClientRect().width || 0);
    for (const menuItem of this._menuItems) {
      if (remainingWidth > 0) {
        const { width } = menuItem.getBoundingClientRect();
        remainingWidth -= width;
        this.setState({ [menuItem.dataset.name || '']: remainingWidth >= 0 });
      } else {
        this.setState({ [menuItem.dataset.name || '']: false });
      }
    }
  };

  _setMenuItemInMap = (node: unknown) => {
    if (node instanceof HTMLElement) this._menuItems.add(node);
  };

  render() {
    const hiddenItems = InterProMin.filter(({ name }) => !this.state[name]);
    return (
      <ul className={css('menu')}>
        {InterProMin.map(({ to, name, activeClass, exact, entities }) => (
          <li
            key={name}
            ref={this._setMenuItemInMap}
            data-name={name}
            className={css('menu-item', { visible: this.state[name] || false })}
            data-testid={`menu-tab-${name.toLowerCase().replace(/\s+/g, '_')}`}
          >
            <MenuItemWithEntities
              to={to}
              activeClass={activeClass}
              exact={exact}
              name={name}
              entities={entities}
            />
          </li>
        ))}
        <ul
          className={css('menu-item', 'view-more', {
            visible: hiddenItems.length,
          })}
          role="tree"
          tabIndex={0}
        >
          <span className={css('more-icon-container')} ref={this._dotdotdotRef}>
            <img src={dotsvg} width="30px" alt="view all menu items" />
          </span>

          {hiddenItems.map(({ to, name, activeClass, exact, entities }) => (
            <li
              key={name}
              className={css('menu-item')}
              role="treeitem"
              aria-selected="false"
            >
              <MenuItemWithEntities
                to={to}
                activeClass={activeClass}
                exact={exact}
                name={name}
                entities={entities}
                nested={true}
              />
            </li>
          ))}
        </ul>
      </ul>
    );
  }
}

export default DynamicMenu;
