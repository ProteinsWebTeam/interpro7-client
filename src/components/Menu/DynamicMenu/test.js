import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import DynamicMenu from '.';

const renderer = new ShallowRenderer();

describe('<DynamicMenu />', () => {
  test('should render', () => {
    renderer.render(<DynamicMenu width={500} />);
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});
