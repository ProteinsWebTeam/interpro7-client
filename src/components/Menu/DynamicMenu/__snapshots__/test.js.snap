// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DynamicMenu /> should render 1`] = `
<ul
  className="menu"
>
  <li
    className="menu-item visible"
    data-name="Home"
    data-testid="menu-tab-home"
  >
    <Memo(Connect(MenuItem))
      exact={true}
      to={
        Object {
          "description": Object {},
        }
      }
    >
      Home
    </Memo(Connect(MenuItem))>
  </li>
  <li
    className="menu-item visible"
    data-name="Search"
    data-testid="menu-tab-search"
  >
    <div
      className="dropdown"
    >
      <Memo(Connect(MenuItem))
        to={
          Object {
            "description": Object {
              "main": Object {
                "key": "search",
              },
            },
          }
        }
      >
        <span
          className="arrow"
        >
          ▾
        </span>
         
        Search
      </Memo(Connect(MenuItem))>
      <ul
        role="tree"
        tabIndex="0"
      >
        <li
          className="menu-item"
        >
          <Memo(Connect(MenuItem))
            activeClass="reactive"
            to={
              Object {
                "description": Object {
                  "main": Object {
                    "key": "search",
                  },
                  "search": Object {
                    "type": "sequence",
                  },
                },
              }
            }
          >
            By Sequence
          </Memo(Connect(MenuItem))>
        </li>
        <li
          className="menu-item"
        >
          <Memo(Connect(MenuItem))
            activeClass="reactive"
            to={
              Object {
                "description": Object {
                  "main": Object {
                    "key": "search",
                  },
                  "search": Object {
                    "type": "text",
                  },
                },
              }
            }
          >
            By Text
          </Memo(Connect(MenuItem))>
        </li>
        <li
          className="menu-item"
        >
          <Memo(Connect(MenuItem))
            activeClass="reactive"
            to={
              Object {
                "description": Object {
                  "main": Object {
                    "key": "search",
                  },
                  "search": Object {
                    "type": "ida",
                  },
                },
              }
            }
          >
            By Domain Architecture
          </Memo(Connect(MenuItem))>
        </li>
      </ul>
    </div>
  </li>
  <li
    className="menu-item visible"
    data-name="Browse"
    data-testid="menu-tab-browse"
  >
    <div
      className="dropdown"
    >
      <Memo(Connect(MenuItem))
        activeClass={[Function]}
        to={[Function]}
      >
        <span
          className="arrow"
        >
          ▾
        </span>
         
        Browse
      </Memo(Connect(MenuItem))>
      <ul
        role="tree"
        tabIndex="0"
      >
        <li
          className="menu-item"
        >
          <Memo(Connect(MenuItem))
            activeClass="reactive"
            to={[Function]}
          >
            By Entry
          </Memo(Connect(MenuItem))>
        </li>
        <li
          className="menu-item"
        >
          <Memo(Connect(MenuItem))
            activeClass="reactive"
            to={[Function]}
          >
            By Protein
          </Memo(Connect(MenuItem))>
        </li>
        <li
          className="menu-item"
        >
          <Memo(Connect(MenuItem))
            activeClass="reactive"
            to={[Function]}
          >
            By Structure
          </Memo(Connect(MenuItem))>
        </li>
        <li
          className="menu-item"
        >
          <Memo(Connect(MenuItem))
            activeClass="reactive"
            to={[Function]}
          >
            By Taxonomy
          </Memo(Connect(MenuItem))>
        </li>
        <li
          className="menu-item"
        >
          <Memo(Connect(MenuItem))
            activeClass="reactive"
            to={[Function]}
          >
            By Proteome
          </Memo(Connect(MenuItem))>
        </li>
        <li
          className="menu-item"
        >
          <Memo(Connect(MenuItem))
            activeClass="reactive"
            to={[Function]}
          >
            By Set
          </Memo(Connect(MenuItem))>
        </li>
      </ul>
    </div>
  </li>
  <li
    className="menu-item visible"
    data-name="Results"
    data-testid="menu-tab-results"
  >
    <div
      className="dropdown"
    >
      <Memo(Connect(MenuItem))
        to={[Function]}
      >
        <span
          className="arrow"
        >
          ▾
        </span>
         
        Results
      </Memo(Connect(MenuItem))>
      <ul
        role="tree"
        tabIndex="0"
      >
        <li
          className="menu-item"
        >
          <Memo(Connect(MenuItem))
            activeClass="reactive"
            to={
              Object {
                "description": Object {
                  "main": Object {
                    "key": "result",
                  },
                  "result": Object {
                    "type": "InterProScan",
                  },
                },
              }
            }
          >
            Your InterProScan searches
          </Memo(Connect(MenuItem))>
        </li>
        <li
          className="menu-item"
        >
          <Memo(Connect(MenuItem))
            activeClass="reactive"
            to={
              Object {
                "description": Object {
                  "main": Object {
                    "key": "result",
                  },
                  "result": Object {
                    "type": "download",
                  },
                },
              }
            }
          >
            Your downloads
          </Memo(Connect(MenuItem))>
        </li>
      </ul>
    </div>
  </li>
  <li
    className="menu-item visible"
    data-name="Release notes"
    data-testid="menu-tab-release_notes"
  >
    <Memo(Connect(MenuItem))
      to={
        Object {
          "description": Object {
            "other": Array [
              "release_notes",
            ],
          },
        }
      }
    >
      Release notes
    </Memo(Connect(MenuItem))>
  </li>
  <li
    className="menu-item visible"
    data-name="Download"
    data-testid="menu-tab-download"
  >
    <Memo(Connect(MenuItem))
      to={
        Object {
          "description": Object {
            "other": Array [
              "download",
            ],
          },
        }
      }
    >
      Download
    </Memo(Connect(MenuItem))>
  </li>
  <li
    className="menu-item visible"
    data-name="Help"
    data-testid="menu-tab-help"
  >
    <div
      className="dropdown"
    >
      <Memo(Connect(MenuItem))
        to={
          Object {
            "description": Object {
              "other": Array [
                "help",
              ],
            },
          }
        }
      >
        <span
          className="arrow"
        >
          ▾
        </span>
         
        Help
      </Memo(Connect(MenuItem))>
      <ul
        role="tree"
        tabIndex="0"
      >
        <li
          className="menu-item"
        >
          <Memo(Connect(MenuItem))
            activeClass="reactive"
            to={
              Object {
                "description": Object {
                  "other": Array [
                    "help",
                    "tutorial",
                  ],
                },
              }
            }
          >
            Tutorials & Webinars
          </Memo(Connect(MenuItem))>
        </li>
        <li
          className="menu-item"
        >
          <Memo(Connect(MenuItem))
            activeClass="reactive"
            to={
              Object {
                "description": Object {
                  "other": Array [
                    "help",
                    "training",
                  ],
                },
              }
            }
          >
            Training
          </Memo(Connect(MenuItem))>
        </li>
        <li
          className="menu-item"
        >
          <Memo(Connect(MenuItem))
            activeClass="reactive"
            to={
              Object {
                "description": Object {
                  "other": Array [
                    "help",
                    "faqs",
                  ],
                },
              }
            }
          >
            FAQs
          </Memo(Connect(MenuItem))>
        </li>
        <li
          className="menu-item"
        >
          <Memo(Connect(MenuItem))
            activeClass="reactive"
            to={
              Object {
                "description": Object {
                  "other": Array [
                    "help",
                    "documentation",
                  ],
                },
              }
            }
          >
            Documentation
          </Memo(Connect(MenuItem))>
        </li>
      </ul>
    </div>
  </li>
  <li
    className="menu-item visible"
    data-name="About"
    data-testid="menu-tab-about"
  >
    <div
      className="dropdown"
    >
      <Memo(Connect(MenuItem))
        to={
          Object {
            "description": Object {
              "other": Array [
                "about",
              ],
            },
          }
        }
      >
        <span
          className="arrow"
        >
          ▾
        </span>
         
        About
      </Memo(Connect(MenuItem))>
      <ul
        role="tree"
        tabIndex="0"
      >
        <li
          className="menu-item"
        >
          <Memo(Connect(MenuItem))
            activeClass="reactive"
            to={
              Object {
                "description": Object {
                  "other": Array [
                    "about",
                    "interpro",
                  ],
                },
              }
            }
          >
            InterPro
          </Memo(Connect(MenuItem))>
        </li>
        <li
          className="menu-item"
        >
          <Memo(Connect(MenuItem))
            activeClass="reactive"
            to={
              Object {
                "description": Object {
                  "other": Array [
                    "about",
                    "interproscan",
                  ],
                },
              }
            }
          >
            InterProScan
          </Memo(Connect(MenuItem))>
        </li>
        <li
          className="menu-item"
        >
          <Memo(Connect(MenuItem))
            activeClass="reactive"
            to={
              Object {
                "description": Object {
                  "other": Array [
                    "about",
                    "consortium",
                  ],
                },
              }
            }
          >
            The InterPro Consortium
          </Memo(Connect(MenuItem))>
        </li>
        <li
          className="menu-item"
        >
          <Memo(Connect(MenuItem))
            activeClass="reactive"
            to={
              Object {
                "description": Object {
                  "other": Array [
                    "about",
                    "funding",
                  ],
                },
              }
            }
          >
            Funding
          </Memo(Connect(MenuItem))>
        </li>
        <li
          className="menu-item"
        >
          <Memo(Connect(MenuItem))
            activeClass="reactive"
            to={
              Object {
                "description": Object {
                  "other": Array [
                    "about",
                    "privacy",
                  ],
                },
              }
            }
          >
            Privacy
          </Memo(Connect(MenuItem))>
        </li>
        <li
          className="menu-item"
        >
          <Memo(Connect(MenuItem))
            activeClass="reactive"
            to={
              Object {
                "description": Object {
                  "other": Array [
                    "about",
                    "team",
                  ],
                },
              }
            }
          >
            Team
          </Memo(Connect(MenuItem))>
        </li>
      </ul>
    </div>
  </li>
  <ul
    className="menu-item view-more"
    role="tree"
    tabIndex="0"
  >
    <span
      className="more-icon-container"
    >
      <img
        alt="view all menu items"
        src={Object {}}
        width="30px"
      />
    </span>
  </ul>
</ul>
`;
